aws/s2n-tls
# fix: fix opened AF_UNIX sockets that didn't call s2n_io_pair_close

### Resolved issues:

Partially solve #4005

### Description of changes: 

* Use s2n_io_pair_close and s2n_io_pair_close_one_end to close most AF_UNIX sockets.
    * Fix those tests that forgot to add s2n_io_pair_close().

### Call-outs:

* See [PR#4834](https://github.com/aws/s2n-tls/pull/4834) and [PR#4835](https://github.com/aws/s2n-tls/pull/4835) for fixes of other open fds.
* I didn't add a test in this PR. The test to detect all opened fds will be in a separate PR.
* Address this [comment](https://github.com/aws/s2n-tls/pull/4833#discussion_r1796098930) in [PR#4841](https://github.com/aws/s2n-tls/pull/4841).

### Testing:

* Test locally. 
    * Add `--track-fds=yes` to CTest memcheck and direct all Valgrind output to `MemoryTester.log` files.
    * Search in VS Code for Open File Descriptors of `AF_UNIX`.
        * Previously opened `AF_UNIX` sockets in tests that are fixed are gone.
    * This is similar to the check in `./codebuild/bin/test_exec_leak.sh`.
  https://github.com/aws/s2n-tls/blob/6bb195c23fd1ecf623dece3b43644e7c9a9e0e20/codebuild/bin/test_exec_leak.sh#L82-L97

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
